### ISSUE TITLE
Please consider porting to the latest PyO3

### ISSUE THREAD

**Issue by musicinmybrain** [milesgranger/cramjam#165]
Please consider porting to the latest PyO3, currently 0.22, which has full support for Python 3.13.

https://pyo3.rs/v0.22.0/migration#from-020-to-021
https://pyo3.rs/v0.22.0/migration#from-021-to-022

**Comment by milesgranger** [comment]
Yeah, I agree. It's in my mental backlog to get done before the next proper release. Just a matter of finding available time. Thanks for putting it into writing though. 😊

**Comment by henryiii** [comment]
Just curious, have you tried using the Stable ABI? That would be a single binary instead one per Python version.

The RC is out now, binaries are supposed to be build for 3.13.

I've been able to work around this by setting `PYO3_USE_ABI3_FORWARD_COMPATIBILITY=1` on CPython 3.13, but that means that it's able to compile with the stable ABI, I believe.

**Comment by milesgranger** [comment]
> have you tried using the Stable ABI? 

If I understand correctly, and I'm just glancing at some of the raw C calls now, the first one I encounter is [PyMemoryView_Check](https://docs.python.org/3/c-api/memoryview.html#c.PyMemoryView_Check) which doesn't appear to be part of the stable ABI, and a few others being used are stable as of 3.11. 

It would've great to only have to do a single binary instead of one per Python version, but that complexity ship has already sailed and for the most part works nicely (outside of trying to also maintain an entirely separate build pipeline for conda despite cramjam having no other Python deps, but that's another story)
